framework/src/media/audio/audio_manager.c

```diff
@@ -514,28 +514,24 @@ static unsigned int resample_stream_out(audio_card_info_t *card, void *data, uns
 	return resampled_frames;
 }
 
-static audio_manager_result_t get_audio_volume(int fd, audio_config_t *config, audio_io_direction_t direct)
+static audio_manager_result_t get_audio_volume(int fd, audio_card_info_t *card, audio_io_direction_t direct)
 {
 	struct audio_caps_desc_s caps_desc;
 	uint8_t max_volume;
 	uint8_t cur_volume;
-	pthread_mutex_t *card_mutex;
 
 	caps_desc.caps.ac_len = sizeof(struct audio_caps_s);
 	caps_desc.caps.ac_type = AUDIO_TYPE_FEATURE;
 
 	if (direct == INPUT) {
 		caps_desc.caps.ac_format.hw = AUDIO_FU_INP_GAIN;
-		card_mutex = &g_audio_in_cards[g_actual_audio_in_card_id].card_mutex;
 	} else {
 		caps_desc.caps.ac_format.hw = AUDIO_FU_VOLUME;
-		card_mutex = &g_audio_out_cards[g_actual_audio_out_card_id].card_mutex;
 	}
 
-	pthread_mutex_lock(card_mutex);
 	if (ioctl(fd, AUDIOIOC_GETCAPS, (unsigned long)&caps_desc.caps) < 0) {
 		meddbg("An ioctl error occurs\n");
-		pthread_mutex_unlock(card_mutex);
+		pthread_mutex_unlock(&card->card_mutex);
```

Comment:
I cannot find where is the 'lock' of this unlock. Is this okay~?

Reply:
This line needs to be removed also, thx!